firebase-firestore/src/main/java/com/google/firebase/firestore/util/Util.java

```diff
@@ -234,4 +240,22 @@ public static int compareByteStrings(ByteString left, ByteString right) {
     }
     return Util.compareIntegers(left.size(), right.size());
   }
+
+  public static List<DocumentKey> extractDocumentKeysFromArrayValue(
```

Comment:
This isn't likely to be a widely used or general utility. Better to make this a package private static member of e.g. `KeyFieldInFilter` and just call it from the other one.

(Another clue that this doesn't belong here is that you're referring to `KeyFieldInFilter` in the error message. This is therefore pretty specific to those classes.)